gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -0,0 +1,190 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2021 Gichan Jang <gichan2.jang@samsung.com>
+ *
+ * @file   tensor_query_common.c
+ * @date   09 July 2021
+ * @brief  Utility functions for tensor query
+ * @see    https://github.com/nnstreamer/nnstreamer
+ * @author Gichan Jang <gichan2.jang@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @see    gsttcpserversink, gsttcpserversrc
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "nnstreamer_log.h"
+#include "tensor_query_common.h"
+
+/** @todo change max read size or need payload */
+#define MAX_READ_SIZE 4096
+
+/**
+ * @brief Create requested socket.
+ * @param[in] hostname the hostname.
+ * @param[in] port a port number.
+ * @param[in] cancellable (nullable) a #GCancellable, or %NULL.
+ * @return Newly created socket or NULL on error.
+ * @note Caller is responsible for unreffiring the returned object with g_object_unref().
+ */
+GSocket *
+gst_tensor_query_socket_new (const gchar * hostname, guint16 port,
+    GCancellable * cancellable)
+{
+  GSocket *socket;
+  GError *err = NULL;
+  GInetAddress *addr;
+  GSocketAddress *saddr;
```

Comment:
Move ```GResolver *resolver;``` to Line 46. It's not referred outside.

Reply:
Moved. Thanks!